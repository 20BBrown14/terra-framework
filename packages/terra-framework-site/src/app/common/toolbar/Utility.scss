/* stylelint-disable selector-class-pattern */
:local {
  .utility {
    align-items: center;
    color: #fff;
    display: flex;
    flex: 0 0 auto;

    & > *:not(:last-child) {
      padding-right: 5px;
    }
  }

  .title {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    word-break: normal;
    word-wrap: normal;
  }
}
