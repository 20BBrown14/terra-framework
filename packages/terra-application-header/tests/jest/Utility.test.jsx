import React from 'react';

import Utility from '../../src/Utility';

describe('Utility', () => {
  it('should render without optional props', () => {
    const result = shallow((
      <Utility />
    ));
    expect(result).toMatchSnapshot();
  });

  it('should render with provided props', () => {
    const result = shallow((
      <Utility
        accessory={<div>test accessory</div>}
        size="small"
        contentHeight="auto"
        contentWidth="auto"
        title="title"
      />
    ));

    expect(result).toMatchSnapshot();
  });
});
