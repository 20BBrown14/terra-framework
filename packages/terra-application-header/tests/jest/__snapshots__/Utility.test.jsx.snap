// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utility should render with provided props 1`] = `
<div>
  <Menu
    contentHeight="auto"
    contentWidth="auto"
    isArrowDisplayed={true}
    isOpen={false}
    onRequestClose={[Function]}
    targetRef={[Function]}
  />
  <Button
    className="utility"
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isReversed={false}
    onClick={[Function]}
    type="button"
    variant="link"
  >
    <div
      className="accessory"
    >
      <div>
        test accessory
      </div>
    </div>
    <div
      className="title"
    >
      title
    </div>
    <IconExpandMore
      className=""
      data-name="Layer 1"
      data-utility-more-icon={true}
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </Button>
</div>
`;

exports[`Utility should render without optional props 1`] = `
<div>
  <Menu
    contentWidth="240"
    isArrowDisplayed={true}
    isOpen={false}
    onRequestClose={[Function]}
    targetRef={[Function]}
  />
  <Button
    className="utility"
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isReversed={false}
    onClick={[Function]}
    type="button"
    variant="link"
  >
    <IconExpandMore
      className=""
      data-name="Layer 1"
      data-utility-more-icon={true}
      isBidi={true}
      viewBox="0 0 48 48"
      xmlns="http://www.w3.org/2000/svg"
    />
  </Button>
</div>
`;
